x-pack/plugins/fleet/server/services/security/message_signing_service.ts

```diff
@@ -135,23 +135,35 @@ export class MessageSigningService implements MessageSigningServiceInterface {
     return publicKey;
   }
 
-  public async rotateKeyPair(): Promise<boolean> {
-    const isRemoved = await this.removeKeyPair();
-    if (isRemoved) {
+  public async rotateKeyPair(): Promise<{ error: Error } | undefined> {
+    try {
+      const removeKeyPairResponse = await this.removeKeyPair();
+      if (!removeKeyPairResponse) {
+        return;
+      }
       await this.generateKeyPair();
-      return true;
+    } catch (error) {
+      return { error: Error(`Error rotating key pair: ${error.message}`) };
```

Comment:
throw instead of return

Reply:
c1976b5a98fec83b21f3218c46ebd8f6f3869340